synthesizer/program/src/logic/instruction/operation/call.rs

```diff
@@ -222,10 +219,16 @@ impl<N: Network> Call<N> {
                 bail!("Expected {} outputs, found {}", closure.outputs().len(), self.destinations.len())
             }
             // Return the output register types.
-            Ok(closure.outputs().iter().map(|output| output.register_type()).cloned().collect())
+            Ok(closure
+                .output_types()
+                .into_iter()
+                // If the function is an external program, we need to qualify its structs or records with
```

Comment:
```suggestion
                // If the function is an external program, we need to qualify its structs
```
Prolly copy pasta, but a closure cannot return a record.